### DISCUSSION TITLE
What models are used to construct the positions of the stars and other bodies?

### DISCUSSION THREAD

**Discussion by Demolishun** [Stellarium/stellarium#3449] [General]
I just started looking at the source code.  I am interested in what kind of models are used to predict the positions of the various bodies in the sky.  Translation: I want to know how it works!  What source files should I start with?  I did a web search about Stellarium to try and understand this, but dug up nothing as to the math involved.  I am assuming it is math heavy.

I searched for model in the discussions but didn't see what I thought would answer my curiosity.

Thanks,
Frank

PS Kind of excited about using this to step forward and backward in time.  That is amazing.

**Comment by gzotti** [comment]
The User Guide lists the relevant sources.
